lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/schema/impl/SchemaContextHolderImpl.java

```diff
@@ -183,27 +194,37 @@ public EffectiveSchemaContext getSchemaContext(final List<GnmiDeviceCapability>
         // Compute schema and add to cache
         final CrossSourceStatementReactor.BuildAction buildAction = yangReactor.newBuild();
         final SchemaException schemaException = new SchemaException();
-        boolean success = true;
-        final Set<GnmiYangModel> completeCapabilities = prepareModelsForSchema(capabilities);
-        for (GnmiYangModel model : completeCapabilities) {
-            try {
-                buildAction.addSource(YangStatementStreamSource.create(makeTextSchemaSource(model)));
-            } catch (IOException | YangSyntaxErrorException e) {
-                LOG.error("Adding YANG {} to reactor failed!", model, e);
-                schemaException.addErrorMessage(e.getMessage());
-                success = false;
+        try {
```

Comment:
Please remove this nested try-catch. This try is here only because prepareModelsForSchema(capabilities) throw exceptions.
I suggest handling exceptions inside the prepareModelsForSchema() or in a separate method like:

```
        ...
        final SchemaException schemaException = new SchemaException();
            boolean success = true;
        final Set<GnmiYangModel> completeCapabilities = getGnmiYangModels(capabilities, schemaException);
        for (GnmiYangModel model : completeCapabilities) {
        ...
```

```
    private Set<GnmiYangModel> getGnmiYangModels(final List<GnmiDeviceCapability> capabilities,
                                                 final SchemaException schemaException)
            throws SchemaException {
        try {
            return prepareModelsForSchema(capabilities);
        } catch (ExecutionException | TimeoutException e) {
            LOG.error("Error reading yang model from datastore", e);
            schemaException.addErrorMessage(e.getMessage());
            throw schemaException;
        } catch (InterruptedException e) {
            LOG.error("Interrupted while reading model from datastore", e);
            Thread.currentThread().interrupt();
            schemaException.addErrorMessage(e.getMessage());
            throw schemaException;
        }
    }
```

Reply:
done